ionics_fits/models/quantum_phys.py

```diff
@@ -44,3 +51,40 @@ def coherent_state_probs(
 
 
 # pytype: enable=invalid-annotation
+
+
+def squeezed_state_probs(
+    n_max: int, zeta: ModelParameter(lower_bound=0)
+) -> Array[("num_fock_states",), np.float64]:
+    """
+    Return occupation probabilities of Fock states for pure squeezed state.
+
+    A pure squeezed state is defined as
+        |ζ> = exp[1 / 2 *  (ζ* a^2 - ζ a_dag^2)] |0>,
+    where a_dag and a denote the harmonic-oscillator creation and annihilation
+    operators.
+
+    The occupation probabilites are truncated at the maximum Fock state |n_max>.
+
+    :param zeta: Complex squeezing parameter
+    """
+    n = np.arange(n_max + 1, dtype=int)
+
+    r = np.abs(zeta)
+    if r == 0:
+        P_n = np.zeros_like(n)
+        P_n[0] = 1
+    else:
+        P_n = (
+            np.power(np.tanh(r), 2 * n)
+            / np.cosh(r)
+            * np.exp(
```

Comment:
do we need to handle potential divide by zero using `np.divide`?

Reply:
cosh(r) doesn't vanish anywhere on the real line